test/distributed/test_functional_api.py

```diff
@@ -704,5 +705,65 @@ def run_with_backward():
         self.assertIsNotNone(t.grad)
 
 
+class TestFunctionalAutogradWithNCCL(MultiProcessTestCase):
```

Comment:
I think this should just inherit from `TestCollectivesWithNCCL` to unify the setups?